src/main/java/com/cburch/logisim/std/arith/AdderHDLGeneratorFactory.java

```diff
@@ -26,10 +26,9 @@
   private static final int NrOfBitsId = -1;
   private static final String ExtendedBitsStr = "ExtendedBits";
   private static final int ExtendedBitsId = -2;
-
-  @Override
-  public String getComponentStringIdentifier() {
-    return "ADDER2C";
+  
+  public AdderHDLGeneratorFactory() {
+    super("ADDER2C");
```

Comment:
Can we have all these strings moved to final static class member, i.e. `_ID` if that makes sense? Multiple places.